src/cpp/editor/events.cpp

```diff
@@ -186,7 +188,7 @@ void InitializeTownEdit(HWND hwnd) {
 	for(int i = 0; i < ELEMENTS_IN(monTypeFields); i++) {
 		SendDlgItemMessage(hwnd, monTypeFields[i], CB_ADDSTRING, 0, (LPARAM)L"-empty-");
 		for(int j = 0; j < GetNumCreatures(); j++) {
-			sprintf(gText, "%s", GetCreatureName(j));
+			snprintf(gText, 64, "%s", GetCreatureName(j));
```

Comment:
Not a bad habit, although I do view this route as pretty hopeless, given the sheer quantity of sprintf in the original code. Where did the number 64 come from?

Reply:
Hmmm. This is essentially assuming that creature names will never grow beyond 64 characters. Probably a safe assumption, but this will be a bug if they do, and one that's impossible to catch.

Why not just use NELEM(gText)?

Reply:
Where is NELEM defined?  I couldn't find it.  I tried `sizeof` but apparently that's not legal on an incomplete type.  If I'm reading the asm correctly, I think the actual size is 300.  I'll use that if you can confirm it for me.

Reply:
https://github.com/jkoppel/project-ironfist/blob/master/src/cpp/shared/base.h#L12

Looks like it's actually called ELEMENTS_IN

Reply:
Hmmm....so, we declare gText as an incomplete type. ELEMENTS_IN uses sizeof, so that won't work either. If you really really want to use snprintf, then this means we'll probably have to move the definition of gText into Ironfist (and make sure there's no code that references intermediate addresses). Is it really worth it?